ports/raspberrypi/common-hal/socketpool/Socket.c

```diff
@@ -745,123 +746,150 @@ socketpool_socket_obj_t *common_hal_socketpool_socket(socketpool_socketpool_obj_
     }
     *incoming_connection = NULL;
 
     // ...and set up the new socket for it.
-    socket2->domain = MOD_NETWORK_AF_INET;
-    socket2->type = MOD_NETWORK_SOCK_STREAM;
-    socket2->incoming.pbuf = NULL;
-    socket2->timeout = socket->timeout;
-    socket2->state = STATE_CONNECTED;
-    socket2->recv_offset = 0;
-    socket2->callback = MP_OBJ_NULL;
-    tcp_arg(socket2->pcb.tcp, (void *)socket2);
-    tcp_err(socket2->pcb.tcp, _lwip_tcp_error);
-    tcp_recv(socket2->pcb.tcp, _lwip_tcp_recv);
+    accepted->domain = MOD_NETWORK_AF_INET;
+    accepted->type = MOD_NETWORK_SOCK_STREAM;
+    accepted->incoming.pbuf = NULL;
+    accepted->timeout = self->timeout;
+    accepted->state = STATE_CONNECTED;
+    accepted->recv_offset = 0;
+    accepted->callback = MP_OBJ_NULL;
+    tcp_arg(accepted->pcb.tcp, (void *)accepted);
+    tcp_err(accepted->pcb.tcp, _lwip_tcp_error);
+    tcp_recv(accepted->pcb.tcp, _lwip_tcp_recv);
 
     tcp_accepted(listener);
 
     MICROPY_PY_LWIP_EXIT
 
+    // output values
+    memcpy(ip, &(accepted->pcb.tcp->remote_ip), NETUTILS_IPV4ADDR_BUFSIZE);
+    *port = (mp_uint_t)accepted->pcb.tcp->remote_port;
+
+    return 1;
+}
+
+socketpool_socket_obj_t *common_hal_socketpool_socket_accept(socketpool_socket_obj_t *socket,
+    uint8_t *ip, uint32_t *port) {
+    // Create new socket object, do it here because we must not raise an out-of-memory
+    // exception when the LWIP concurrency lock is held
+    socketpool_socket_obj_t *accepted = m_new_ll_obj_with_finaliser(socketpool_socket_obj_t);
+    socketpool_socket_reset(accepted);
+
+    int ret = socketpool_socket_accept(socket, ip, port, accepted);
+
+    if (ret <= 0) {
+        m_del_obj(socketpool_socket_obj_t, accepted);
+        if (ret == 0) {
+            // Interrupted.
+            return mp_const_none;
+        }
+        mp_raise_OSError(-ret);
+    }
+
     DEBUG_printf("registering socket in socketpool_socket_accept()\n");
-    if (!register_open_socket(socket2)) {
+    if (!register_open_socket(accepted)) {
         DEBUG_printf("collecting garbage to open socket\n");
         gc_collect();
-        if (!register_open_socket(socket2)) {
+        if (!register_open_socket(accepted)) {
             mp_raise_RuntimeError(translate("Out of sockets"));
         }
     }
-    mark_user_socket(socket2);
+    mark_user_socket(accepted);
 
-    // output values
-    memcpy(ip, &(socket2->pcb.tcp->remote_ip), NETUTILS_IPV4ADDR_BUFSIZE);
-    *port = (mp_uint_t)socket2->pcb.tcp->remote_port;
-    return MP_OBJ_FROM_PTR(socket2);
+    return MP_OBJ_FROM_PTR(accepted);
 }
 
 bool common_hal_socketpool_socket_bind(socketpool_socket_obj_t *socket,
     const char *host, size_t hostlen, uint32_t port) {
 
     // get address
     ip_addr_t bind_addr;
-    int error = socketpool_resolve_host(socket->pool, host, &bind_addr);
-    if (error != 0) {
-        mp_raise_OSError(EHOSTUNREACH);
-    }
+    const ip_addr_t *bind_addr_ptr = &bind_addr;
```

Comment:
was this causing a compiler diagnostic or something?

Reply:
No, I added support for empty host "" to bind to all IPs of the interface. It also doesn't need to run a resolve then.